### PR TITLE
Mitch branch

### PR DESCRIPTION
**Pull request by mqharris**

### CODE REVIEW

File: app_layout_1.R [hunk 1]
```diff
@@ -45,13 +48,28 @@ list(
...
-htmlBr())))}
-   # add radio button dependent drop down,
-  
+htmlBr(),
+dccDropdown(
+  id = "radio-dependent",
+  options=list(),
+  multi=TRUE,
+  value=NULL,
+  # labelStyle=list("display"="block")
```
Comment (mqharris) [inline app_layout_1.R:69]: this exists in the python repo

but if i include it here, it gives me a "param not used"

File: src/app/app_wrangling.R [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+# return unique values to populate dropdowns
+subset_data <- function(data, col_ = "category") {
+    func_df_out <- data %>% unnest(.data[[col_]])
+
+    unique_out <- map(func_df_out[[col_]], ~ (unlist(.x))) %>%
+        unlist() %>%
+        unique() %>%
+        na.omit()
+
+    return(list(unique_out))
```
Comment (mqharris) [inline src/app/app_wrangling.R:188]: i changed `return(unique_out)` to `return(list(unique_out))`

If i dont make the change it looks like i only get one element back

i.e the drop down boxes on the first tab only have 1 thing listed

File: app_layout_1.R [hunk 3]
```diff
@@ -7,19 +7,22 @@ library(dashCoreComponents)
...
-
-
+boardgame_data <- call_boardgame_data()
 
 # * WRANGLING - get dictionary for dropdowns here
+col_key_list <- c("category", "mechanic", "publisher")
+col_dict <- vector(mode="list", length=3)
+names(col_dict) <- col_key_list
 
+for (idx in 1:3){
```
Comment (mqharris) [inline app_layout_1.R]: sub optimal, couldn't figure out a map right away
ill keep working on it
and im open to suggestions ; )

Comment (WraySmith) [reply app_layout_1.R]: lets get it all working into the `dasjboard_layout` branch and then do clean-up before into `main`, for loop all good at this level

File: app_layout_1.R (hunk 1 above)
Comment (WraySmith) [reply app_layout_1.R:69]: `# labelStyle=list("display"="block")` you mean that line?

File: app_layout_1.R [hunk 4]
```diff
@@ -203,8 +221,17 @@ selected_style=tab_selected_style)))))))
...
 
 # radio button selection options to populate drop down
-
 # this will return something like [{"label": c, "value": c} for c in col_dict[col]]
+app$callback(
+  list(output("radio-dependent", "options")),
+  list(input("radio-selection", "value")),
+  function(chosen_selection){
+
+    outlist <- list(lapply(col_dict[[chosen_selection]], function(x) return(list(label=x, value=x))))
```
Comment (WraySmith) [inline app_layout_1.R:230]: lets stick with using the `purrr` package - something that can be cleaned up before the layout branch goes into main

**Review by WraySmith — APPROVED** [review]
Some things still need to be cleaned up: removing a few `for` loops and switching `lapply()` to `map` for consistency with the wrangling file but I think should just be merged into the working `sophia_branch`. Just finish linting and merge in.

File: app_layout_1.R (hunk 1 above)
Comment (mqharris) [reply app_layout_1.R:69]: yeah

File: app_layout_1.R (hunk 4 above)
Comment (mqharris) [reply app_layout_1.R:230]: kk

**Pull request merged**